### ISSUE TITLE
Quarto vignettes use the installed version of the package, but Rmd vignettes use the current dev version

### ISSUE THREAD

**Issue by venpopov** [r-lib/pkgdown#2830]
I just found out you can have Quarto vignettes and to try them out, I created a new package. I added a quarto vignette, but when I tried to build the site with `build_site()`, I got the following error:

```
── Installing package mytestpkg into temporary library ───────────────────────────────────────────
Error: 
! in callr subprocess.
Caused by error in `quarto::quarto_render(path, metadata_file = metadata_path, execute_dir = output_dir, …`:
! ✖ Error running quarto cli.
ℹ Rerun with `quiet = FALSE` to see the full error message.
Caused by error:
! System command 'quarto' failed
ℹ See `$stderr` for standard error.
Type .Last.error to see the more details.
```

Rendering regular .Rmd vignettes was fine. I was finally able to track down the issue: the code `library(mytestpkg)` in the vignette failed for the Quarto vignette, because in contrast to the Rmd vignette, it did not use see the temporary library.

Ok, so I `install()` the package locally and then it renders successfully. The problem is that if you make any updates, and you rebuild the website, the Quarto vignette will be built with the locally installed version, while the Rmd vignettes will be built with the current version that `build_site()` installs to a temporary library.

![Screenshot 2024-12-07 at 07 11 58](https://github.com/user-attachments/assets/48acfb8f-984f-4a41-b8cc-c4ff4d2e8fe4)
![Screenshot 2024-12-07 at 07 12 13](https://github.com/user-attachments/assets/5063af5e-7caa-4856-a085-de7dd77ad66d)

The deployment via github pages workflow doesn't have this problem and everything works fine, but locally Rmd and Qmd vignettes are built differently.

**Comment by hadley** [comment]
Hmmm, probably need to set `R_LIBS` to the temporary directory that we install the package into so that quarto uses it.

**Comment by Robinlovelace** [comment]
This may be related or the same: https://github.com/quarto-dev/quarto-actions/issues/120

**Comment by Robinlovelace** [comment]
Update: @cderv confirmed that the above issue is the same as this. It is a bug that hinders use of .qmd files for vignettes and, unlike @venpopov's statement above, we found it can lead to error messages in Actions. I would say this is fairly high priority, anything I can do to help with diagnostics/fix testing let me know :pray:

**Comment by cderv** [comment]
> Hmmm, probably need to set R_LIBS to the temporary directory that we install the package into so that quarto uses it.

I agree with this. @jayhesselberth I can help with that. I'll try to do a PR to fix this.

**Comment by jayhesselberth** [comment]
@cderv If is something that will come up frequently maybe this could also be adressed in the quarto R package with a `lib_dir` param for `quarto_render()`?

**Comment by cderv** [comment]
I am not sure it will be often, but surely in some context like pkgdown when 
- A package is installed in a temp directory added to current session `.libPath()`
- `quarto_render()` is run with expectation of the library being available, which won't be the case if `R_LIBS` was not modified and passed to the new process. 

**pkgdown** being the one to install in a temp directory, it would make sense for it to be sure this libs is known to the new process. 

Thouhg, I wonder if this can create some issue with vignette engine too, so I'll check.

**Comment by b-rodrigues** [comment]
Just was confronted to the same issue, glad to see it's being worked on :+1:

**Comment by cortinah** [comment]
Perhaps related, R CMD build in Github Actions fails for my hockeystick package where I just added a Quarto vignette. this only happens on Windows, ubuntu and macos are fine.
Hope this is useful. thanks

Run r-lib/actions/check-r-package@v2
Run ## --------------------------------------------------------------------
── R CMD build ─────────────────────────────────────────────────────────────────
* checking for file 'D:\a\hockeystick\hockeystick/DESCRIPTION' ... OK
* preparing 'hockeystick':
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... ERROR
Error: --- re-building 'usinghockeystick.qmd' using html
processing file: usinghockeystick.qmd
1/25               
2/25 [carbon]      
Error in `library()`:
! there is no package called 'hockeystick'

**Comment by jayhesselberth** [comment]
@cortinah your issue is likely unrelated, pkgdown includes some test quarto vignettes and builds on all platforms.

**Comment by cortinah** [comment]
> [@cortinah](https://github.com/cortinah) your issue is likely unrelated, pkgdown includes some test quarto vignettes and builds on all platforms.

ok, thanks!

**Comment by davidkane9** [comment]
Any updates on this? Any suggestions for how to deal with it in the meantime? I just switched my vignettes from Rmd to qmd.  I think that the above issue is giving me a [similar failure](https://github.com/PPBDS/tutorial.helpers/actions/runs/13337209955/job/37254954525).

**Comment by jayhesselberth** [comment]
@davidkane9 can you rebuild your site after installing this branch?

```r
pak::pak("r-lib/pkgdown@quarto-libpath")
```

**Comment by davidkane9** [comment]
Install worked:

```
> packageVersion("pkgdown")
[1] ‘2.1.1.9000’
>
```

But I get a new error:

```
> pkgdown::build_site()
── Installing package tutorial.helpers into temporary library ────────────────────────────────────
── Initialising site ───────────────────────────────────────────────────────────
Copying <pkgdown>/BS5/assets/katex-auto.js to katex-auto.js
Copying <pkgdown>/BS5/assets/lightswitch.js to lightswitch.js
Copying <pkgdown>/BS5/assets/link.svg to link.svg
...
Writing [articles/index.html](vscode-file://vscode-app/Applications/Positron.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#)
Reading vignettes/addins.qmd
Reading vignettes/books.qmd
Reading vignettes/cloud.qmd
Reading vignettes/downloads.qmd
Reading vignettes/instructions.qmd
Reading vignettes/testing.qmd
Running `quarto render`
 
Error:
! ! in callr subprocess.
Caused by error in `quarto::quarto_render(path, metadata_file = metadata_path, execute_dir = output_dir, …`:
! ✖ Error running quarto cli.
ℹ Rerun with `quiet = FALSE` to see the full error message.
Caused by error:
! System command 'quarto' failed
Hide Traceback
    ▆
 1. └─pkgdown::build_site()
 2.   └─pkgdown:::build_site_external(...)
 3.     └─callr::r(...)
 4.       └─callr:::get_result(output = out, options)
 5.         └─throw(callr_remote_error(remerr, output), parent = fix_msg(remerr[[3]]))
>
```

Any advice much appreciated.

**Comment by jayhesselberth** [comment]
@cderv I wonder if you had made any inroads in tracking this error down? I have not been able to come up with a simple reprex package that triggers this error on GHA. 

https://jayhesselberth.github.io/pkgdown.quarto.issue

**Comment by Robinlovelace** [comment]
Any ideas when this fix will be in the CRAN version? Thanks!

**Comment by cderv** [comment]
I don't think the pkgdown PR that closed this will solve the problem. This issue is related to 
- quarto-dev/quarto-r#217

If a package is installed in a temporary directory added to `.libPaths()` in the current session, the Quarto subprocess won't know about it as the R sessions will be independant. 

Setting `install = FALSE` and `new_process = FALSE` should show no issue in quarto rendering. 

I have tried quarto-dev/quarto-r#228, which solves part of the problem. I am still puzzled about the behavior on Windows. What the PR does in **quarto** is set `.libPaths()` values to `R_LIBS` env var so that when Quarto CLI starts R, `R_LIBS` is considered and the same .libPaths() is used. 

If anyone wants to try quarto-dev/quarto-r#228, please do so, and report back in the PR. thank you!

**Comment by jayhesselberth** [comment]
@venpopov @Robinlovelace we are trying to figure this out but it is a confusing interaction between pkgdown and quarto.

The current question is: what OS you are using? At least one of the issues is windows-specific.

**Comment by Robinlovelace** [comment]
Ubuntu locally for sure and Ubuntu on the remote machine in the actions workflow I believe

**Comment by cderv** [comment]
Let me try to sum up: Quarto vignettes are used in Two context

- R CMD build / check => in that context there is a windows only problem because of some behavior in CI with windows R. This is about
	- https://github.com/quarto-dev/quarto-r/issues/217

- pkgdown vignettes & articles rendering => in that context, when `install = TRUE` (the default in `pkgdown::build_site()` you could run locally), the dev package you are trying to build the doc site for will be installed in a temporary directory. And this is not working with CRAN quarto because this temporary directory is not known / seen by quarto and its R sub process. It impact all OSes and the fix is 
	- quarto-dev/quarto-r#228 

I hope this is clearer.

**Comment by francisbarton** [comment]
>     * pkgdown vignettes & articles rendering => in that context, when `install = TRUE` (the default in `pkgdown::build_site()` you could run locally), the dev package you are trying to build the doc site for will be installed in a temporary directory. And this is not working with CRAN quarto because this temporary directory is not known / seen by quarto and its R sub process. It impact all OSes and the fix is
>       
>       * [Pass `.libPaths()` as R_LIBS to quarto subprocess quarto-dev/quarto-r#228](https://github.com/quarto-dev/quarto-r/pull/228)
>       
>       Until this PR is merged, the package needs to be installed first in the default library so that it is found when quarto renders the .qmd vignette.
> 

@cderv am I right in thinking that we have no control over this when relying on the pkgdown GitHub Action to render quarto vignettes? We just have to wait for the PR to get merged and then go to CRAN and then the fix will eventually be in the version used by the GitHub Ubuntu runner?

Or is there some workaround to use your PR branch? I am not fully understanding if there is a fix that can be implemented now or if we just have to wait.
